File: RaceTrackerServer/src/main/java/com/fpvracetracker/model/frontend/RoundResultEntry.java
package com.fpvracetracker.model.frontend;

import java.util.Date;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
@Entity
@Table(name = "roundResultEntries")
public class RoundResultEntry extends BaseEntity {

	private static final long serialVersionUID = -6437552894066134177L;
	public String pilotUUID;
	public String pilotNumber;
	public String pilotName;
	public int rank;
	@ElementCollection(fetch = FetchType.EAGER)
	@Column(name = "lapTimes")
//	@Type(type = "com.model.ByteArrayType")
	public List<Double> lapTimes;
	public double lapTimesSum;
	public String deviceId;
	public boolean disqualified;

	@JsonIgnore
	@ManyToOne
	public Race race;

	public RoundResultEntry() {
		super();
	}

	public RoundResultEntry(String uuid, Date saved, Date synced, String pilotUUID, String pilotNumber,
			String pilotName, int rank, List<Double> lapTimes, double lapTimesSum, String deviceId,
			boolean disqualified, boolean deleted) {
		super(uuid, saved, synced, deleted);
		this.pilotUUID = pilotUUID;
		this.pilotNumber = pilotNumber;
		this.pilotName = pilotName;
		this.rank = rank;
		this.lapTimes = lapTimes;
		this.lapTimesSum = lapTimesSum;
		this.deviceId = deviceId;
		this.disqualified = disqualified;
	}

	public String getPilotUUID() {
		return pilotUUID;
	}

	public void setPilotUUID(String pilotUUID) {
		this.pilotUUID = pilotUUID;
	}

	public String getPilotNumber() {
		return pilotNumber;
	}

	public void setPilotNumber(String pilotNumber) {
		this.pilotNumber = pilotNumber;
	}

	public String getPilotName() {
		return pilotName;
	}

	public void setPilotName(String pilotName) {
		this.pilotName = pilotName;
	}

	public int getRank() {
		return rank;
	}

	public void setRank(int rank) {
		this.rank = rank;
	}

	public List<Double> getLapTimes() {
		return lapTimes;
	}

	public void setLapTimes(List<Double> lapTimes) {
		this.lapTimes = lapTimes;
	}

	public double getLapTimesSum() {
		return lapTimesSum;
	}

	public void setLapTimesSum(double lapTimesSum) {
		this.lapTimesSum = lapTimesSum;
	}

	public String getDeviceId() {
		return deviceId;
	}

	public void setDeviceId(String deviceId) {
		this.deviceId = deviceId;
	}

	public boolean isDisqualified() {
		return disqualified;
	}

	public void setDisqualified(boolean disqualified) {
		this.disqualified = disqualified;
	}

}

File: RaceTrackerServer/src/main/resources/web/script/js/racetrackermodel/RoundResultEntry.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var RoundResultEntry = (function (_super) {
    __extends(RoundResultEntry, _super);
    function RoundResultEntry(json) {
        //TODO: evaluate if roundresultentry could be replaced with raceresultentry
        _super.call(this, json);
        this.pilotUUID = json.pilotUUID;
        this.rank = json.rank;
        this.lapTimes = json.lapTimes ? json.lapTimes : [];
        this.lapTimesSum = json.lapTimesSum;
        this.pilotName = json.pilotName;
        this.pilotNumber = json.pilotNumber;
        this.deviceId = json.deviceId;
        this.disqualified = json.disqualified ? json.disqualified : false;
    }
    return RoundResultEntry;
}(BaseEntity));
